knowage-core/src/main/java/it/eng/spagobi/api/DossierActivityResource.java

```diff
@@ -195,29 +186,27 @@ public Response importTemplateFile(MultiPartBody multipartFormDataInput) throws
 		byte[] archiveBytes = null;
 		JSONObject response = new JSONObject();
 		try {
-			String separator = File.separator;
 			final FormFile file = multipartFormDataInput.getFormFileParameterValues("file")[0];
 			ParameterValue[] documentIdArray = multipartFormDataInput.getParameteValues("documentId");
 			String identifier = "";
-			String path = null;
+
+			String fileName = file.getFileName();
+			archiveBytes = file.getContent();
+
+			File f = null;
 			if (documentIdArray.length == 1) {
 				identifier = documentIdArray[0].toString();
-				path = SpagoBIUtilities.getResourcePath() + separator + "dossier" + separator + identifier + separator;
+				f = PathTraversalChecker.get(SpagoBIUtilities.getResourcePath(), "dossier", identifier, fileName);
 			} else {
 				identifier = multipartFormDataInput.getParameteValues("uuid")[0].toString();
-				path = Files.createTempDirectory("prefix").getParent().resolve(identifier).toString() + separator;
+				f = PathTraversalChecker.get(SpagoBIUtilities.getResourcePath(), "dossier", identifier, fileName);
```

Comment:
this should be
PathTraversalChecker.get(System.getProperty("java.io.tmpdir"), identifier)